### PR TITLE
Migrate `border` custom properties from `v10` to `v11`

### PR DESCRIPTION
**Pull request by lgriffee**
## WHY are these changes introduced?

Part of #8405 

## WHAT is this pull request doing?

Runs the [`v11-styles-replace-custom-property-border`](https://github.com/Shopify/polaris/blob/780acafd681ffeba6f5df634f290bee646557b64/polaris-migrator/src/migrations/v11-styles-replace-custom-property-border/v11-styles-replace-custom-property-border.ts) from `@shopify/polaris-migrator` to migrate `border` custom properties from `v10` to `v11`. e.g.

```sh
cd polaris-react
npx @shopify/polaris-migrator v11-styles-replace-custom-property-border './src/**/*.scss'
npx prettier --write "./src/**/*.scss"
yarn lint:styles
```

This PR changes the `BorderAliasTokens` in the `Divider` component. As a result, there are slight visual changes to the `Divider` component border colors. Please see screenshots of documentation changes below: 

### polaris.shopify.com 
**Before**
Screenshot 2023-04-11 at 11 40 06 AM

**After**  
Screenshot 2023-04-11 at 11 48 52 AM

#### Storybook 
**Before** 
Screenshot 2023-04-11 at 11 51 44 AM

**After** 
Screenshot 2023-04-11 at 11 40 26 AM

### CODE REVIEW

**Review by aveline — APPROVED** [review]

**Review by aaronccasanova — APPROVED** [review]

**Review by laurkim — COMMENTED** [review]
The `Divider` component passes in `divider` token in as a string [default arg](https://github.com/Shopify/polaris/blob/703eaa888d04d708ef8e01007de86798d23e30d4/polaris-react/src/components/Divider/Divider.tsx#L17), so I think it wasn't caught in the migrations. 

This can probably be updated something along the lines of the below:
```suggestion
export const Divider = ({borderStyle = 'border-subdued'}: DividerProps) => {
  const style = {
    '--pc-divider-border-style': borderStyle
      ? `var(--p-border-width-1) solid var(--p-color-${borderStyle})`
      : undefined,
  } as React.CSSProperties;
```
with `BorderTokenAlias` updated to accept `border-subdued` instead of `divider` as a type.
I think this will resolve one of the failing Divider tests for CI check.

For the other failing test, we could update `dark` as a type for `BorderTokenAlias` to `border` and then the test should pass. This might require all of the border token aliases to be updated though to reflect the changes.

**Comment by lgriffee** [comment]
@laurkim Thanks so much for catching those legacy `border` token instances inside the `Divider `component! And for your suggestions on how to fix it. ❤️🙏🏻 I've pushed up changes in 03e4d93 that I think should fix this issue! Alongside this, I've added screenshots in the PR description showing the resulting visual changes to the doc site and Storybook.

**Review by laurkim — APPROVED** [review]

**Review by alex-page — APPROVED** [review]
Hyped lets go!

File: polaris-react/src/components/Divider/Divider.stories.tsx [hunk 1]
```diff
@@ -14,21 +14,17 @@ export function WithBorderStyles() {
...
       <Text as="h1" variant="headingXs">
-        Divider
+        Border inverse
       </Text>
-      <Divider borderStyle="divider" />
-      <Text as="h1" variant="headingXs">
-        Divider on dark
-      </Text>
-      <Divider borderStyle="divider-on-dark" />
+      <Divider borderStyle="border-inverse" />
```
Comment (alex-page) [inline polaris-react/src/components/Divider/Divider.stories.tsx]: @lgriffee @laurkim could we undo the changes to `<Divider>` here and do a larger breaking change as seen in this PR? https://github.com/Shopify/polaris/pull/8909
